index.js

```diff
@@ -243,6 +246,58 @@ module.exports = function(version, _options) {
             }
 
             return output;
+        },
+        getBestMatchingLanguage: function(language) {
+            if (languages.instructions[language]) return language;
+
+            // If the language is not found, try a little harder
+            var splitLocale = language.toLowerCase().split('-');
+            var languageCode = false;
+            var scriptCode = false;
+            var countryCode = false;
+
+            if (splitLocale.lenth === 1) {
+                languageCode = splitLocale[0];
+            } else if (splitLocale.length === 2 && splitLocale[1].length === 4) {
+                languageCode = splitLocale[0];
+                scriptCode = splitLocale[1];
+            } else if (splitLocale.length === 2 && splitLocale[1].length === 2) {
+                languageCode = splitLocale[0];
+                countryCode = splitLocale[1];
+            } else if (splitLocale.length === 3) {
+                languageCode = splitLocale[0];
+                scriptCode = splitLocale[1];
+                countryCode = splitLocale[2];
```

Comment:
Here’s [a useful reference](https://en.wikipedia.org/wiki/IETF_language_tag#Syntax_of_language_tags) on the syntax of a BCP47 language tag. We’re currently handling all the parts that a client is likely to pass in.